crates/revm/revm-inspectors/src/tracing/builder/parity.rs

```diff
@@ -218,13 +238,156 @@ impl ParityTraceBuilder {
+                Some(child) => {
+                    child_idx_stack.push(child_idx + 1);
+
+                    child_idx = 0;
+                    current = self.nodes.get(*child).expect("there should be a child");
+                }
+                None => {
+                    let mut instructions: Vec<VmInstruction> =
+                        Vec::with_capacity(current.trace.steps.len());
+
+                    for step in &current.trace.steps {
+                        let maybe_sub = match step.op.u8() {
+                            opcode::CALL
+                            | opcode::CALLCODE
+                            | opcode::DELEGATECALL
+                            | opcode::STATICCALL
+                            | opcode::CREATE
+                            | opcode::CREATE2 => {
+                                sub_stack.pop_front().expect("there should be a sub trace")
+                            }
+                            _ => None,
+                        };
+
+                        instructions.push(Self::make_instruction(step, maybe_sub));
+                    }
+
+                    match current.parent {
+                        Some(parent) => {
+                            sub_stack.push_back(Some(VmTrace {
+                                code: Default::default(),
+                                ops: instructions,
+                            }));
+
+                            child_idx = child_idx_stack.pop().expect("there should be a child idx");
+
+                            current = self.nodes.get(parent).expect("there should be a parent");
+                        }
+                        None => break instructions,
+                    }
+                }
+            }
+        };
+
+        VmTrace { code: Default::default(), ops: instructions }
+    }
+
+    /// todo::n config
+    ///
+    /// Creates a VM instruction from a [CallTraceStep] and a [VmTrace] for the subcall if there is one
+    fn make_instruction(step: &CallTraceStep, maybe_sub: Option<VmTrace>) -> VmInstruction {
+        let maybe_storage = match step.storage_change {
+            Some(storage_change) => {
+                Some(StorageDelta { key: storage_change.key, val: storage_change.value })
+            }
+            None => None,
+        };
+
+        let maybe_memory = match step.memory.len() {
+            0 => None,
+            _ => {
+                Some(MemoryDelta { off: step.memory_size, data: step.memory.data().clone().into() })
+            }
+        };
+
+        let maybe_execution = Some(VmExecutedOperation {
+            used: step.gas_cost,
+            push: match step.new_stack {
+                Some(new_stack) => Some(new_stack.into()),
+                None => None,
+            },
+            mem: maybe_memory,
+            store: maybe_storage,
+        });
+
+        VmInstruction {
+            pc: step.pc,
+            cost: 0, // todo::n
```

Comment:
what to do here?

this is the gas cost of the opcode?
we should be able to get this via `spec_opcode_gas`

I think we need to get the spec id from the env during tracing.

I can add this in a follow up